ui/desktop/tests/acceptance/projects/sessions/session-test.js

```diff
@@ -160,7 +164,29 @@ module('Acceptance | projects | sessions | session', function (hooks) {
     });
 
     await visit(urls.target);
-    await click('[data-test-target-detail-connect-button]');
+    await click(TARGET_CONNECT_BUTTON);
+
+    assert.strictEqual(currentURL(), urls.session);
+  });
+
+  test('visiting a session that does not have read permissions but a successful connect', async function (assert) {
+    assert.expect(2);
+    // verifies second call to sessions/:id is also a 403
+    QUnit.onUncaughtException = (err) => {
+      assert.true(err.errors[0].isForbidden);
+    };
```

Comment:
To give a little explanation as to why this was necessary:

We [re-retrieve](https://github.com/hashicorp/boundary-ui/blob/e5263ba52c910cc7f19402f184bc81b5e2cefaab/ui/desktop/app/routes/scopes/scope/projects/sessions/session.js#L23) the session model when we enter the sessions route. We either previously preloaded by manually pushing to the store or successfully retrieved the model during connection. This retrieval will first pull from the cache in ember data store and _then_ retrieve the model again in the background (unless we set `backgroundReload: false`). 

This background refresh is the issue as it will still return a `403` but we can't catch it anywhere as it happens in the background inside ember internals. So eventually the test will fail without anyway for us to catch the error.

We _could_ turn off the background refresh as we don't really currently use any data that could be updated; the only thing that would get updated is the state of the session (i.e. when a session goes from pending to active once we connect). But we never really turn it off in our application and I think it could be unexpected in the future if we ever did need it, the only real negative to this would be in the situation where a user doesn't have permissions, we would be making an extra call that would fail silently in the background.

Reply:
Resolving so I can merge.